tests/ctrl/bootstrap.go

```diff
@@ -6,99 +6,54 @@ import (
 	"fmt"
 	"io/ioutil"
 	"log"
-	"net/http"
 	"time"
 
 	"github.com/Inphi/eip4844-interop/shared"
 	"github.com/ethereum/go-ethereum/core"
-	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/ethereum/go-ethereum/params"
 	types "github.com/prysmaticlabs/prysm/v3/consensus-types/primitives"
 	beaconservice "github.com/prysmaticlabs/prysm/v3/proto/eth/service"
 	ethpbv1 "github.com/prysmaticlabs/prysm/v3/proto/eth/v1"
-	"google.golang.org/grpc"
+	"golang.org/x/sync/errgroup"
 	"gopkg.in/yaml.v2"
 )
 
-func WaitForService(ctx context.Context, url string) error {
-	ctx, cancel := context.WithTimeout(ctx, time.Second*60)
-	defer cancel()
-	for {
-		if _, err := http.Get(url); err == nil {
-			break
-		}
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		case <-time.After(1 * time.Second):
-		}
-	}
-	return nil
-
-}
-
-func WaitForGeth(ctx context.Context) error {
-	log.Printf("waiting for geth")
-	return WaitForService(ctx, shared.GethRPC)
-}
+var env *TestEnvironment
 
-func WaitForBeaconNode(ctx context.Context) error {
-	log.Printf("waiting for prysm beacon node")
-	return WaitForService(ctx, fmt.Sprintf("%s/eth/v1/beacon/genesis", fmt.Sprintf("http://%s", shared.BeaconGatewayGRPC)))
-}
-
-func WaitForBeaconNodeFollower(ctx context.Context) error {
-	log.Printf("waiting for prysm beacon node follower")
-	return WaitForService(ctx, fmt.Sprintf("%s/eth/v1/beacon/genesis", shared.BeaconFollowerRPC))
-}
-
-func WaitForValidator(ctx context.Context) error {
-	log.Printf("waiting for validator")
-	return WaitForService(ctx, shared.ValidatorRPC)
-}
-
-func WaitForServices(ctx context.Context) error {
-	if err := WaitForGeth(ctx); err != nil {
-		return fmt.Errorf("%w: geth offlinev", err)
-	}
-	if err := WaitForBeaconNode(ctx); err != nil {
-		return fmt.Errorf("%w: beacon node offline", err)
-	}
-	if err := WaitForBeaconNodeFollower(ctx); err != nil {
-		return fmt.Errorf("%w: beacon node follower offline", err)
-	}
-	if err := WaitForValidator(ctx); err != nil {
-		return fmt.Errorf("%w: validator is offline", err)
+func InitE2ETest() {
+	ctx := context.Background()
+	if err := StopDevnet(); err != nil {
+		log.Fatalf("unable to stop devnet: %v", err)
 	}
-	return nil
+	env := GetEnv()
+	env.StartAll(ctx)
 }
 
-var Env *TestEnvironment
-
-func InitE2ETest() {
-	if err := RestartDevnet(); err != nil {
```

Comment:
Do we still need the RestartDevnet() function at all now that you remove this call? Seems to be the only call site.

Reply:
and then you could also get rid of StartDevnet() too I think.

Reply:
You're right. Will remove.